WalletWasabi.Tests/UnitTests/UserInterfaceTest/PocketSelectionTests.cs

```diff
@@ -0,0 +1,402 @@
+		Assert.Contains(selection.GetLabel("Roland"), selection.LabelsBlackList);
+		Assert.Contains(selection.GetLabel("Dan"), selection.LabelsBlackList);
+
+		selection.GetLabel("Dan").IsPointerOver = true;
+
+		Assert.True(selection.GetLabel("Dan").IsHighlighted);
+		Assert.False(selection.GetLabel("Roland").IsHighlighted);
+		Assert.False(selection.GetLabel("Target").IsHighlighted);
+
+		var output = selection.GetUsedPockets();
+
+		Assert.DoesNotContain(pocket1, output);
+		Assert.DoesNotContain(pocket2, output);
+	}
+
+	[Fact]
+	public void BlackListHighlightsDealWithMultipleOverlapsCorrectly()
+	{
+		var selection = new LabelSelectionViewModel(Money.Parse("1.0"));
+
+		var pockets = new List<Pocket>();
+		pockets.AddPocket(1.0M, out var pocket1, "Target", "Dan");
+		pockets.AddPocket(1.0M, out var pocket2, "Target");
+		pockets.AddPocket(1.0M, out var pocket3, "Target", "Roland");
+
+		selection.Reset(pockets.ToArray());
+
+		selection.SwapLabel(selection.GetLabel("Target"));
+
+		Assert.Contains(selection.GetLabel("Target"), selection.LabelsBlackList);
+		Assert.Contains(selection.GetLabel("Roland"), selection.LabelsBlackList);
+		Assert.Contains(selection.GetLabel("Dan"), selection.LabelsBlackList);
+
+		selection.GetLabel("Dan").IsPointerOver = true;
+
+		Assert.True(selection.GetLabel("Dan").IsHighlighted);
+		Assert.False(selection.GetLabel("Roland").IsHighlighted);
+		Assert.True(selection.GetLabel("Target").IsHighlighted);
+
+		var output = selection.GetUsedPockets();
+
+		Assert.DoesNotContain(pocket1, output);
+		Assert.DoesNotContain(pocket2, output);
+		Assert.DoesNotContain(pocket3, output);
+	}
+
+	[Fact]
+	public void WhiteListHighlightsDealWithMultipleOverlapsCorrectly()
+	{
+		var selection = new LabelSelectionViewModel(Money.Parse("1.0"));
+
+		var pockets = new List<Pocket>();
+		pockets.AddPocket(1.0M, out var pocket1, "Target", "Dan");
+		pockets.AddPocket(1.0M, out var pocket2, "Target");
+		pockets.AddPocket(1.0M, out var pocket3, "Target", "Roland");
+
+		selection.Reset(pockets.ToArray());
+
+		selection.GetLabel("Dan").IsPointerOver = true;
+
+		Assert.True(selection.GetLabel("Dan").IsHighlighted);
+		Assert.False(selection.GetLabel("Roland").IsHighlighted);
+		Assert.False(selection.GetLabel("Target").IsHighlighted);
+	}
+
+	[Fact]
+	public void WhiteListHighlightsGroupedLabelsInOtherPocketsThatContainTargetLabelExceptThoseAvailableInOtherPockets()
+	{
+		var selection = new LabelSelectionViewModel(Money.Parse("1.0"));
+
+		var pockets = new List<Pocket>();
+		pockets.AddPocket(1.0M, out var pocket1, "Target", "Dan");
+		pockets.AddPocket(1.0M, out var pocket2, "Target");
+		pockets.AddPocket(1.0M, out var pocket3, "Target", "Roland");
+
+		selection.Reset(pockets.ToArray());
+
+		selection.GetLabel("Dan").IsPointerOver = true;
+
+		Assert.False(selection.GetLabel("Target").IsHighlighted);
+		Assert.False(selection.GetLabel("Roland").IsHighlighted);
+		Assert.True(selection.GetLabel("Dan").IsHighlighted);
+	}
```

Comment:
This test exactly testing the same logic as `WhiteListHighlightsDealWithMultipleOverlapsCorrectly`